test/test_symmetry.cpp

```diff
@@ -0,0 +1,39 @@
+#include <gtest/gtest.h>
+
+extern "C" {
+#include "cell.h"
+#include "symmetry.h"
+#include "utils.h"
+}
+
+TEST(test_symmetry, test_get_lattice_symmetry_layer) {
+    Cell *cell;
+    Symmetry *symmetry;
+
+    cell = NULL;
+    symmetry = NULL;
+
+    const int size = 1;
+    double lattice[3][3] = {
+        {1, 0, 0},
+        {0, 1, 0},
+        {0, 0, 1},
+    };
+    double positions[1][3] = {{0, 0, 0}};
+    const int types[1] = {0};
+    const int aperiodic_axis = 2;
+    const double symprec = 1e-5;
+    const double angle_tolerance = -1;
+
+    cell = cel_alloc_cell(size, NOSPIN);
+    cel_set_layer_cell(cell, lattice, positions, types, aperiodic_axis);
+
+    // Bravais group of the two-dimensional lattice is 4/mmm
+    symmetry = sym_get_operation(cell, symprec, angle_tolerance);
+    ASSERT_EQ(symmetry->size, 16);
+
+    sym_free_symmetry(symmetry);
+    cel_free_cell(cell);
+    symmetry = NULL;
+    cell = NULL;
```

Comment:
`NULL` -> `nullptr`. Also you mean to assert it not assign it here.

Reply:
At least for me,  I am not familiar with C++. If we want to write in C, is there any good way to do it?

Reply:
I would say that writing in C++, especially for these tests it is more advantageous because:
- you get a clear trace of how/what fails
- you decouple potential C memory issues, e.g. not pre-allocating pointers from tests result errors

In these case I only have 2 simple suggestions:
- Use [`std::array`](https://en.cppreference.com/w/cpp/container/array) and continue to use the C interface as you would.
  The reason for this is that you allow in the future to use a wider range of built-in tools like range-for loops similar to python, that make the code much easier to read and detangle.
- Check with the IDE's suggestion.

The `NULL` -> `nullptr` is suggested by CLion and that is because the `NULL` definition in C is different then C++'s `nullptr` and C++ uses the latter for enforcing that a pointer is truely null and not a pointer assigned to `(void*)0`. Think of the issues of comparing addresses of empty strings (`&"" == &""`). That is not guaranteed to give `true` and depends on the compiler's optimization.

Other than that there is a genuine error here in that it is supposed to be `ASSERT_EQ(symmetry == nullptr)` and not an assignment `symmetry = nullptr`.

Reply:
I see your point. But I want to confine this PR about the algorithmic side for the layer-group implementation for now. After we adopt C11 in develop branch, the corresponding refactoring would be awesome.